src/app/components/languages/languages.component.ts

```diff
@@ -0,0 +1,31 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Language} from '../../models/language';
+import {LanguageService} from '../../service/language.service';
+
+@Component({
+  selector: 'languages',
+  templateUrl: './languages.component.html'
+})
+export class LanguagesComponent implements OnInit {
+  @Input() name: string;
+  @Input() code: string;
+  languages: Language[];
+
+  constructor(private languageService: LanguageService) {}
+
+  ngOnInit(): void {
+    this.languageService.getLanguages().then(languages => this.languages = languages);
+  }
+
+  createLanguage(): void {
+    const l: Language = new Language();
+    l.name = this.name;
+    l.code = this.code;
+
+    this.languageService.createLanguage(l).then(result => {
+
```

Comment:
Refresh? This seems to have no action. Maybe you just haven't implemented that yet.